src/pages/products/Products.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { FaTrashAlt, FaBatteryThreeQuarters, FaCheck, FaEdit, FaShoppingBag } from 'react-icons/fa';
+
+import { getProductsByCategory } from '../../services';
+import InputSpan from '../../components/form';
+import * as Constants from '../../constants';
+
+import './products.scss';
+
+class Products extends Component {
+
+    state = {
+        products: null,
+        displayFilter: []
+    };
+
+    insideRefs = [];
+
+    setRef = (numberOfRefs) => {
+        this.insideRefs = [...Array(numberOfRefs)]
+            .map((inputSpan) => inputSpan = React.createRef())
+    };
+
+    componentDidMount() {
+        getProductsByCategory()
+            .then(products => {
+                this.setState({ products: products.products });
+            })
+    }
+
+    deleteTask = () => {
+        console.log('deleted')
+    }
+
+    simulateClick(index) {
+        this.insideRefs[index].current.handleSpanBlur();
+    }
+
+    handleInputChange = (e) => {
+        if (e.target.value.length > 2) {
+            let displayFilter = this.state.products.filter(product => product.title.includes(e.target.value));
+            displayFilter = displayFilter.length ? displayFilter : Constants.NO_RESULTS;
+            this.setState({ displayFilter: displayFilter })
+
+        } else {
+            this.setState({ displayFilter: [] })
+        }
+        return true
+    }
+
+    handleInputSpanChange = (id, e) => {
+        let { products } = this.state;
+        products = this.state.products.map(product => (
+            {
+                id: product.id,
+                title: product.id === id ? e.target.value : product.title
+            }
+        ));
+        this.setState({ products: products });
+    }
+
+    render() {
+
+        const { products, displayFilter } = this.state;
+        const productItems = displayFilter.length ? displayFilter : products;
+        products && this.setRef(products.length);
+
+        return (
+            <div className='component-products'>
+                <label>{ Constants.FILTER }</label>
+                <input onChange={ this.handleInputChange } type="text"/>
+                <span>{ Constants.MORE_SYMBOLS }</span>
+                <div className="products-wrapper">
+                    { displayFilter === Constants.NO_RESULTS ?
+                        <div className='product-list'>{ Constants.NO_RESULTS }</div> :
+                        productItems && (
+                            Object.values(productItems).map((productList, index) => (
+                                    <div className="product-wrapper" key={ productList.id }>
```

Comment:
I would move `product-wrapper` in a separate component. Keep in mind for the future, we need to avoid very comlex or big component if possible. Here it is possible